File: CORREDOR/Tipos Constructores/Go/Perro.go
package main

import "fmt"

type Perro struct {
	Animal
	Edad int
}
//Constructor Con Parametros
func NewPerro(nombre, raza string, edad int) Perro {
	return Perro{
		Animal: NewAnimal(nombre, raza),
		Edad:   edad,
	}
}

func (d Perro) HacerSonido() {
	fmt.Println("El perro ladra: ¡Guau guau!")
}

func (d Perro) Informacion() {
	fmt.Println("Nombre:", d.Nombre)
	fmt.Println("Raza:", d.Raza)
	fmt.Println("Edad:", d.Edad)
}

File: CORREDOR/Tipos Constructores/C#/MiProyecto/Perro.cs
public class Perro : Animal, AnimalActions{
    public int Edad { get; set; }
    //Constructor con parametros
    public Perro(string nombre, int edad) : base(nombre){
        Edad = edad;
    }
    public void Sonido(){
        Console.WriteLine("¡Guau!");
    }
    public override void Informacion(){
        Console.WriteLine($"Nombre del perro: {Nombre}");
        Console.WriteLine($"Edad: {Edad}");
    }
}